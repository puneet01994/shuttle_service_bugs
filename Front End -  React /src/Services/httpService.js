import axios from "axios";
import SessionStorageService from "./SessionStorageService";

import AlertTypes from "../Redux/actions/AlertTypes";

axios.defaults.baseURL = "http://d302d9ca93b6.ngrok.io";

const sessionStorageService = SessionStorageService.getService();
const isHandlerEnabled = (config = {}) => {
  // eslint-disable-next-line no-prototype-builtins
  return config.hasOwnProperty("handlerEnabled") && !config.handlerEnabled
    ? false
    : true;
};


axios.interceptors.request.use((request) => requestHandler(request));

const requestHandler = (request) => {
  const at = sessionStorageService.getAccessToken();

  if (request) {
    request.headers["Authorization"] = at;
  }
  return request;
};

axios.interceptors.response.use(
  (response) => successHandler(response),
  (error) => errorHandler(error)
);

const successHandler = (response) => {
  if (isHandlerEnabled(response.config)) {
    // AlertTypes(response.status);
  }
  return response;
};

export const errorHandler = (error) => {
  if (isHandlerEnabled(error.config)) {
    AlertTypes(
      error.response && error.response.status,
      error.message,
      error.config
    );

    return Promise.reject({ ...error });
  }
};

export default {
  get: axios.get,
  put: axios.put,
  post: axios.post,
  delete: axios.delete,
};
