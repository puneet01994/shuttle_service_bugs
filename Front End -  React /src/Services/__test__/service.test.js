import { newAcceessToken, errorHandler, successHandler } from "../httpService";
import axios from "axios";
import moxios from "moxios";
import swal from "sweetalert";
jest.mock("sweetalert");

jest.mock("axios");

test("should fetch users", () => {
  const token = {
    payload: "",
  };
  const originalRequest = "";

  const resp = { data: token, status: 200 };
  axios.post.mockResolvedValue(resp);

  axios.post.mockResolvedValue(resp);

  return newAcceessToken(originalRequest).then((data) =>
    expect("").toEqual(resp.data.payload)
  );
});

test("error handler", () => {
  const error = {
    response: {
      config: {
        url: "/Authorize",
        _retry: true,
      },
      data: {
        status: 401,
        message: "Unauthorized User",
        path: "/admin/addlocation",

        details: ["Enter area name with company name"],
      },
      status: 401,
    },
  };

  const abc = errorHandler(error);
  expect(abc).toMatchSnapshot();
});
