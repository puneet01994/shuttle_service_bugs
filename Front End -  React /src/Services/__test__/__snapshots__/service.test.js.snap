// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error handler 1`] = `undefined`;

exports[`error handler 2`] = `undefined`;
